plugins/outputs/datadog/datadog.go

```diff
@@ -220,6 +238,20 @@ func verifyValue(v interface{}) bool {
 	return true
 }
 
+func isRateable(metricType string, fieldName string) bool {
```

Comment:
We have the use of "metricType" in a couple places, can you add a comment to this function that explains or specifies that we are looking for metric type from statsd tag for specific cases?

Reply:
I think explicitly naming the variable `statsDMetricType` would help too.

```go
// Retrieve the metric_type tag created by inputs.statsd
statsDMetricType, _ := m.GetTag("metric_type")
# ...
if d.ShouldRateCounts && isRateable(statsDMetricType, fieldName) {
# ...

func isRateable(statsDMetricType string, fieldName string) bool {
	switch statsDMetricType {
	case
		"counter":
		return true
	case
		"timing",
		"histogram":
		return fieldName == "count"
	default:
		return false
	}
}
```

I can then maybe put some better documentation in the README to explicitly say only `inputs.statsd` is supported. How does that sound?

Reply:
That looks great and makes it much clearer, thank you!

Reply:
Done in https://github.com/influxdata/telegraf/pull/15702/commits/c80846351828646a971d5763485fb965e6e1784b